core/tmparchive/tmparchive.rb

```diff
@@ -3,42 +3,93 @@
 require_relative '../header.rb'
 
 # ---------------------- VARIABLES
+local_log_hash, @log_hash = Bkmkr::Paths.setLocalLoghash
+
 input_config = File.join(Bkmkr::Paths.submitted_images, "config.json")
 
 tmp_config = File.join(Bkmkr::Paths.project_tmp_dir, "config.json")
 
 # ---------------------- METHODS
+## all methods for this script are Mcmlln::Tools methods wrapped in new methods,
+## in order to return results for json_logfile
+def getFilesinSubmittedImages(logkey='')
+	files = Mcmlln::Tools.dirList(Bkmkr::Paths.submitted_images)
+	files
+rescue => logstring
+	return []
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
 
-# ---------------------- PROCESSES
+def makeFolder(path, logkey='')
+	unless Dir.exist?(path)
+		Mcmlln::Tools.makeDir(path)
+	else
+	 logstring = 'n-a'
+	end
+rescue => logstring
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
 
-# For TEST purposes
-test_images_before = Mcmlln::Tools.dirList(Bkmkr::Paths.submitted_images)
+def deleteOldProjectTmpFolder(logkey='')
+	if Dir.exist?(Bkmkr::Paths.project_tmp_dir)
+		Mcmlln::Tools.deleteDir(Bkmkr::Paths.project_tmp_dir)
+	else
+		logstring = 'n-a'
+	end
+rescue => logstring
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
 
-# Local path variables
-all_submitted_images = Mcmlln::Tools.dirList(Bkmkr::Paths.submitted_images)
+def copyInputFile(logkey='')
+	Mcmlln::Tools.copyFile("#{Bkmkr::Project.input_file}", Bkmkr::Paths.project_tmp_file)
+rescue => logstring
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
 
-# Rename and move input files to tmp folder to eliminate possibility of overwriting
-unless Dir.exist?(Bkmkr::Paths.tmp_dir)
-	Mcmlln::Tools.makeDir(Bkmkr::Paths.tmp_dir)
+def mvInputConfigFile(input_config, logkey='')
+	if File.file?(input_config)
+		Mcmlln::Tools.moveFile(input_config, tmp_config)
+	else
+		logstring = 'n-a'
+	end
+rescue => logstring
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
 end
 
-if Dir.exist?(Bkmkr::Paths.project_tmp_dir)
-	Mcmlln::Tools.deleteDir(Bkmkr::Paths.project_tmp_dir)
+def writeAlertFile(filecontents, logkey='')
+	Mcmlln::Tools.overwriteFile(Bkmkr::Paths.alert, filecontents)
+rescue => logstring
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
 end
 
-Mcmlln::Tools.makeDir(Bkmkr::Paths.project_tmp_dir)
+# ---------------------- PROCESSES
+# Local path variables
+all_submitted_images = getFilesinSubmittedImages('check_submitted_images')
+# log submitted image list
+@log_hash['submitted_images'] = all_submitted_images
+
+# Rename and move input files to tmp folder to eliminate possibility of overwriting
```

Comment:
can you move this comment to immediately precede line 86? It was in a weird place in the previous incarnation of the code, so not something you introduced, but would be nice to fix since it is a bit confusing.